ambry-account/src/test/java/com/github/ambry/account/HelixAccountServiceTest.java

```diff
@@ -479,13 +498,34 @@ public void testConflictingUpdateCaseE() throws Exception {
     // write two accounts (1, "a") and (2, "b")
     writeAccountsForConflictTest();
     Collection<Account> conflictAccounts =
-        Collections.singleton((new AccountBuilder((short) 1, "b", AccountStatus.INACTIVE, null).build()));
+        Collections.singleton((new AccountBuilder((short) 1, "b", AccountStatus.INACTIVE).build()));
     assertFalse("Wrong return value from update operation.", accountService.updateAccounts(conflictAccounts));
     assertEquals("Wrong account number in HelixAccountService", 2, accountService.getAllAccounts().size());
     assertEquals("Wrong account name got from HelixAccountService", "a",
         accountService.getAccountById((short) 1).getName());
   }
 
+  /**
+   * Test updating an account with a conflicting expected snapshot version.
+   */
+  @Test
+  public void testConflictingSnapshotVersionUpdate() throws Exception {
+    accountService = mockHelixAccountServiceFactory.getAccountService();
+    // write two accounts (1, "a") and (2, "b")
+    writeAccountsForConflictTest();
+    Account expectedAccount = accountService.getAccountById((short) 1);
+    for (int snapshotVersion : new int[]{-1, 0, 2}) {
+      Collection<Account> conflictAccounts = Collections.singleton(
+          new AccountBuilder((short) 1, "a", AccountStatus.INACTIVE).snapshotVersion(snapshotVersion).build());
+      assertFalse("Wrong return value from update operation.", accountService.updateAccounts(conflictAccounts));
+      assertEquals("Wrong account number in HelixAccountService", 2, accountService.getAllAccounts().size());
+      assertEquals("Account should not have been updated", expectedAccount, accountService.getAccountById((short) 1));
```

Comment:
Might be missing something but would it make sense to try to rename this account to "c" when creating `conflictAccounts` on line 518? Right now, even if the update somehow went through, since the old props are the same as the new ones, the equals wouldn't fail, right?

Reply:
That's a good idea, I will change another property as well. I did not modify this method after I removed the snapshot version from the equals check.